src/openms/source/ANALYSIS/TOPDOWN/DeconvolutedSpectrum.cpp

```diff
@@ -0,0 +1,456 @@
+      const double intensity = pg.getIntensity();
+
+      auto crange = pg.getChargeRange();
+
+      fs << fileName << "\t" << pg.getScanNumber() << "\t"
+         << std::to_string(spec.getRT()) << "\t"
+         << size() << "\t"
+         << std::to_string(am) << "\t" << std::to_string(m) << "\t" << intensity << "\t"
+         << std::get<0>(crange) << "\t" << std::get<1>(crange) << "\t"
+         << pg.size() << "\t";
+
+      if (writeDetail)
+      {
+        fs << std::fixed << std::setprecision(2);
+        for (auto &p : pg)
+        {
+          fs << p.mz << ";";
+        }
+        fs << "\t";
+
+        fs << std::fixed << std::setprecision(1);
+        for (auto &p : pg)
+        {
+          fs << p.intensity << ";";
+        }
+        fs << "\t";
+        fs << std::setprecision(-1);
+
+        for (auto &p : pg)
+        {
+          fs << p.charge << ";";
+        }
+        fs << "\t";
+        for (auto &p : pg)
+        {
+          fs << p.getUnchargedMass() << ";";
+        }
+        fs << "\t";
+        for (auto &p : pg)
+        {
+          fs << p.isotopeIndex << ";";
+        }
+        fs << "\t";
+
+        for (auto &p : pg)
+        {
+          auto tm = pg.getMonoMass() + p.isotopeIndex * Constants::ISOTOPE_MASSDIFF_55K_U;
+          auto diff = (tm / abs(p.charge) + FLASHDeconvHelperStructs::getChargeMass(p.charge > 0) - p.mz) / p.mz;
+          fs << 1e6 * diff << ";";
+        }
+        fs << "\t";
+      }
+      if (spec.getMSLevel() > 1)
+      {
+        //PrecursorScanNum	PrecursorMz	PrecursorIntensity PrecursorCharge	PrecursorMonoMass		PrecursorQScore
+        fs << precursorScanNumber << "\t" << std::to_string(precursorPeak.getMZ()) << "\t"
+           << precursorPeak.getIntensity() << "\t"
+           << precursorPeak.getCharge()
+           << "\t";
+
+        if (precursorPeakGroup == nullptr)
+        {
+          fs << "nan\tnan\t";
+        }
+        else
+        {
+          fs << std::to_string(precursorPeakGroup->getMonoMass()) << "\t"
+             << precursorPeakGroup->getQScore() << "\t";
+        }
+      }
+      fs << pg.getIsotopeCosine() << "\t" << pg.getChargeScore() <<"\t";
+
+      auto qrange = pg.getMzxQScoreMzRange();
+      fs << pg.getSNR() << "\t"
+         << pg.getRepCharge() << "\t" << std::to_string(std::get<0>(qrange)) << "\t"
+         << std::to_string(std::get<1>(qrange)) << "\t"
+         << pg.getQScore() << "\t" << std::setprecision(-1); //
+
+      for (int i = std::get<0>(crange); i <= std::get<1>(crange); i++)
+      {
+
+        fs << pg.getChargeIntensity(i);
+
+        if (i < std::get<1>(crange))
+        {
+          fs << ";";
+        }
+      }
+      fs << "\t";
+      int isoEndIndex = 0;
+
+      for (auto &p : pg)
+      {
+        isoEndIndex = isoEndIndex < p.isotopeIndex ? p.isotopeIndex : isoEndIndex;
+      }
+      auto perIsotopeIntensity = new double[isoEndIndex + 1];
```

Comment:
1. use std::array or std::vector but please no C array. Why? because you forgot to delete it ;)

Reply:
Man shame on me.